app/views/courses/TeacherClassView.coffee

```diff
@@ -448,9 +450,11 @@ module.exports = class TeacherClassView extends RootView
     stats.totalPlaytime = if playtime then moment.duration(playtime, "seconds").humanize() else 0
     # TODO: Humanize differently ('1 hour' instead of 'an hour')
 
-    levelPracticeMap = {}
-    levelPracticeMap[level.id] = level.get('practice') ? false for level in @levels.models
-    completeSessions = @classroom.sessions.filter (s) -> s.get('state')?.complete and not levelPracticeMap[s.get('levelID')]
+    levelIncludeMap = {}
+    language = @classroom.get('aceConfig')?.language
+    for level in @levels.models
+      levelIncludeMap[level.get('original')] = not level.get('practice') and (not language? or level.get('primerLanguage') isnt language)
+    completeSessions = @classroom.sessions.filter (s) -> s.get('state')?.complete and levelIncludeMap[s.get('level')?.original]
```

Comment:
(continued from app/models/Classroom.coffee)
And here, if we want @levels filtered in every context it's used, probably call the filter function on it just after it's created.